frontend/src/metabase/user/components/UserSettings.jsx

```diff
@@ -12,13 +12,37 @@ import UserAvatar from "metabase/components/UserAvatar";
 import SetUserPassword from "./SetUserPassword";
 
 export default class UserSettings extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { updateUserResult: null };
+  }
+
   static propTypes = {
     tab: PropTypes.string.isRequired,
     user: PropTypes.object.isRequired,
     setTab: PropTypes.func.isRequired,
     updatePassword: PropTypes.func.isRequired,
   };
 
+  onUserSettingsSaved(details) {
+    this.setState({
```

Comment:
Likewise this would become


```js
this.setState({
  formError: null,
  formSuccess: { data: { message: ... } }
})
```

and

```js
this.setState({
  formError: { data: { message: ... } },
  formSuccess: null
})
```